desc/objectives/objective_funs.py

```diff
@@ -740,14 +663,57 @@ def weights(self):
             [jnp.ones(obj.dim_f) * obj.weight for obj in self.objectives]
         )
 
+    @property
+    def _all_things(self):
+        """list: all things known to this objective, used and unused."""
+        if not hasattr(self, "_extra_things"):
+            self._extra_things = []
+        return [obj.things for obj in self.objectives] + self._extra_things
```

Comment:
extra_things just refers to unused things by this objective right?

Reply:
yes, for example in a constrained problem the objective and constraints may depend on different objects, but we want both to accept the same set of arguments. This is effectively the newer version of the old `align_jacobian` function that @ddudt was asking about